test/Dialect/Verif/lower-formal-to-hw.mlir

```diff
@@ -0,0 +1,11 @@
+// RUN: circt-opt --lower-formal-to-hw %s | FileCheck %s
+
+hw.module @Foo(in %bar : i32, in %baz : i16) {}
+
+// CHECK-LABEL: hw.module @FormalTop(in %symbolic_value_0 : i32, in %symbolic_value_1 : i16)
+verif.formal @FormalTop {
+  %0 = verif.symbolic_value : i32
+  %1 = verif.symbolic_value : i16
+  // CHECK-NEXT: hw.instance "foo" @Foo(bar: %symbolic_value_0: i32, baz: %symbolic_value_1: i16)
+  hw.instance "foo" @Foo(bar: %0: i32, baz: %1: i16) -> ()
+}
```

Comment:
Can you try adding a `comb` and `seq` op in here somewhere as well? I think the dialect conversion framework might fail in that case because the comb/seq dialects aren't marked as legal.

Reply:
Yea, are there any other dialects besides comb and seq we want to include?

Reply:
Hmmm, I'm wondering if we can get around listing the dialects explicitly. Since you just have a rewrite pattern that you want to apply exhaustively wherever it can, you could use the greedy pattern rewriter instead of the dialect conversion. That doesn't try to be clever about which ops are legal or illegal; it just applies the pattern everywhere.

This should work:
- Make your pattern inherit from `OpRewritePattern`. (You probably have to move the `make_early_inc_range` down to the loop where you actually call `eraseOp`, because this now applies more eagerly than with dialect conversion.)
- Use `applyPatternsAndFoldGreedily(getOperation(), std::move(patterns))` instead of the dialect conversion framework. (This needs `mlir/Transforms/GreedyPatternRewriteDriver.h`.)
- That should allow you to get rid of the conversion target entirely.

The greedy rewriter should then just apply your pattern until no more `verif.formal` ops are left.